tests/certification/bindings/cron/cron_test.go

```diff
@@ -0,0 +1,223 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cron_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/components-contrib/bindings/cron"
+	bindings_loader "github.com/dapr/dapr/pkg/components/bindings"
+
+	"github.com/dapr/dapr/pkg/runtime"
+	"github.com/dapr/kit/logger"
+
+	dapr_testing "github.com/dapr/dapr/pkg/testing"
+	"github.com/stretchr/testify/require"
+
+	daprClient "github.com/dapr/go-sdk/client"
+	"github.com/dapr/go-sdk/service/common"
+
+	// Certification testing runnables
+	"github.com/dapr/components-contrib/tests/certification/embedded"
+	"github.com/dapr/components-contrib/tests/certification/flow"
+	"github.com/dapr/components-contrib/tests/certification/flow/app"
+	"github.com/dapr/components-contrib/tests/certification/flow/sidecar"
+	"github.com/dapr/components-contrib/tests/certification/flow/watcher"
+)
+
+// Test cron trigger with most frequent schedule possible : @every 1s with a context deadline
+func TestCronBindingFrequentTrigger(t *testing.T) {
+	ports, _ := dapr_testing.GetFreePorts(3)
+	grpcPort := ports[0]
+	httpPort := ports[1]
+	appPort := ports[2]
+
+	triggerWatcher := watcher.NewOrdered()
+	for i := 1; i <= 10; i++ {
+		triggerWatcher.ExpectInts(i)
+	}
+
+	// test if cron triggers 10 times within 10 seconds
+	testAssert := func(ctx flow.Context) error {
+		triggerWatcher.Assert(ctx, time.Second*10)
+		return nil
+	}
+
+	application := func(ctx flow.Context, s common.Service) error {
+		triggeredCount := 0
+		// Setup the input binding endpoint
+		err := s.AddBindingInvocationHandler("cron", func(_ context.Context, in *common.BindingEvent) ([]byte, error) {
+			ctx.Logf("Cron triggered at %s", time.Now().String())
+			triggeredCount++
+			triggerWatcher.Observe(triggeredCount)
+			return []byte("{}"), nil
+		})
+		require.NoError(t, err)
+		return nil
+	}
+
+	testDelete := func(ctx flow.Context) error {
+		client := sidecar.GetClient(ctx, "cron-sidecar")
+
+		req := &daprClient.InvokeBindingRequest{Name: "cron", Operation: "delete"}
+		res, err := client.InvokeBinding(ctx, req)
+
+		require.Equal(t, res.Metadata["schedule"], "@every 1s")
+		require.NotEmpty(t, res.Metadata["stopTimeUTC"])
+		require.NoError(t, err, "Error cancelling cron schedule")
+		return nil
+	}
+
+	flow.New(t, "test cron trigger schedule @every 1s").
+		Step(app.Run("cronapp", fmt.Sprintf(":%d", appPort), application)).
+		Step(sidecar.Run("cron-sidecar",
+			embedded.WithComponentsPath("./components"),
+			embedded.WithDaprGRPCPort(grpcPort),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprHTTPPort(httpPort),
+			componentRuntimeOptions(),
+		)).
+		Step("assert cron triggered within deadline", testAssert).
```

Comment:
I am wondering if this will be flaky, since we need to test it in exactly 10 seconds. If it gets delayed by a bit and another cron is triggered, then assertion will fail. Increasing the trigger time can help with it.

Reply:
100% agree. Actually, please consider replacing the timer in the Cron binding with one that allows faking time. Just measuring time is going to lead to very flaky tests (we see that all the time in the runtime). See: https://github.com/benbjohnson/clock
Then, in tests you can control time passing by with the mock clock

Reply:
This would require forking `github.com/robfig/cron/v3` used for cron binding
This PR was never merged https://github.com/robfig/cron/pull/327

Reply:
Sounds good to me. Let's fork it, since it's not actively maintained.

Looks like the code of the package is pretty short, so you may just import the code into our tree by creating a sub-package internal/cron or something like that (just make sure you report the original license correctly!)